server/consumer.go

```diff
@@ -1503,21 +1508,30 @@ func stopAndClearTimer(tp **time.Timer) {
 
 // Stop will shutdown  the consumer for the associated stream.
 func (o *Consumer) Stop() error {
-	return o.stop(false, true)
+	return o.stop(false, true, false)
 }
 
-// Delete will delete the consumer for the associated stream.
+func (o *Consumer) deleteWithoutAdvisory() error {
+	return o.stop(true, true, false)
+}
+
+// Delete will delete the consumer for the associated stream and send advisories.
 func (o *Consumer) Delete() error {
-	return o.stop(true, true)
+	return o.stop(true, true, true)
 }
 
-func (o *Consumer) stop(dflag, doSignal bool) error {
+func (o *Consumer) stop(dflag, doSignal, advisory bool) error {
 	o.mu.Lock()
 	mset := o.mset
 	if mset == nil {
 		o.mu.Unlock()
 		return nil
 	}
+
+	if advisory {
+		o.sendDeleteAdvisoryLocked()
```

Comment:
stop here does not always mean delete. IIRC when you shutdown a server but expect to restore the consumer state on restart this is called with dflag false. So maybe just need

```go
if dflag && advisory {}
```